diff --git a/node_modules/@capacitor-community/apple-sign-in/ios/Sources/SignInWithApple/Plugin.swift b/node_modules/@capacitor-community/apple-sign-in/ios/Sources/SignInWithApple/Plugin.swift
index f53087c..89f202b 100755
--- a/node_modules/@capacitor-community/apple-sign-in/ios/Sources/SignInWithApple/Plugin.swift
+++ b/node_modules/@capacitor-community/apple-sign-in/ios/Sources/SignInWithApple/Plugin.swift
@@ -1,6 +1,7 @@
 import Foundation
 import Capacitor
 import AuthenticationServices
+import UIKit
 
 @objc(SignInWithApple)
 public class SignInWithApple: CAPPlugin, CAPBridgedPlugin {
@@ -24,6 +25,7 @@ public class SignInWithApple: CAPPlugin, CAPBridgedPlugin {
 
         let authorizationController = ASAuthorizationController(authorizationRequests: [request])
         authorizationController.delegate = self
+        authorizationController.presentationContextProvider = self
         authorizationController.performRequests()
     }
 
@@ -83,3 +85,12 @@ extension SignInWithApple: ASAuthorizationControllerDelegate {
         self.bridge?.releaseCall(call)
     }
 }
+
+extension SignInWithApple: ASAuthorizationControllerPresentationContextProviding {
+    public func presentationAnchor(for controller: ASAuthorizationController) -> ASPresentationAnchor {
+        if let bridge = self.bridge, let viewController = bridge.viewController, let window = viewController.view.window {
+            return window
+        }
+        return UIApplication.shared.windows.first ?? UIWindow()
+    }
+}
